Rosella/src/main/java/me/hydos/rosella/util/SemaphorePool.java

```diff
@@ -0,0 +1,98 @@
+package me.hydos.rosella.util;
+
+import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
+import org.lwjgl.system.MemoryStack;
+import org.lwjgl.vulkan.VK10;
+import org.lwjgl.vulkan.VkDevice;
+import org.lwjgl.vulkan.VkSemaphoreCreateInfo;
+
+import java.nio.LongBuffer;
+import java.util.Set;
+import java.util.Stack;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class SemaphorePool {
+
+    private static final int ALLOCATION_SIZE = 16;
```

Comment:
is there a reason behind this number? just curious. its not a problem

Reply:
No not really i just had to pick some random number and 16 looked nice.